torch/csrc/jit/passes/quantization.cpp

```diff
@@ -630,23 +643,41 @@ class InsertQuantDeQuantHelper {
   std::tuple<c10::QScheme, QParamMap> getQSchemeAndQParamMap(
       script::Module& module,
       Node* n);
+  void checkQScheme(Graph* g, c10::QScheme qscheme) {
+    if (qscheme_for_graph_.count(g)) {
+      TORCH_CHECK(toAffine(qscheme_for_graph_.at(g)) == toAffine(qscheme) ||
+
+                  "Quantizing same graph with different types of "
+                  "QSchemes is not supported.\n",
+                  " Expecting:",
+                  c10::toString(qscheme_for_graph_.at(g)),
+                  " Got:",
+                  c10::toString(qscheme));
+    } else {
+      qscheme_for_graph_[g] = qscheme;
```

Comment:
Since we're comparing with `toAffine(qscheme)`, we should probably store `toAffine(qscheme)` rather than plain `qscheme`. Otherwise the check would fail for, e.g, `c10::kPerTensorSymmetric`.

Reply:
you mean moving the toAffine call in comparison here right?